### ISSUE TITLE
aiohttp client sessions are often not properly closed

### ISSUE THREAD

**Issue by hoh** [aleph-im/aleph-client#125]
The aiohttp session management should use context managers instead of the current caching.

This creates a lot of errors about "Unclosed client session" in logs of daemons.

**Comment by MHHukiewitz** [comment]
I'm using a sort of client object here: https://github.com/aleph-im/active-record-sdk/blob/main/src/aars/core.py#L262
Would this pattern work for what you intend?

**Comment by hoh** [comment]
The following approach is currently supported:
```python3
with aiohttp.ClientSession() as session:
    messages = aleph_client.synchronous.get_messages(session=session, ...)
```

We could either enforce passing a session, but that requires users to know about aiohttp and we don't want to expose that all the time.

Synchronous functions could open and close the session in the synchronous wrapper, solving the issue for some usecases (unless a session object is passed, then it is the responsibility of the caller).

We could also provide our own wrapper, something like:
```python3
with aleph_client.synchronous.Client() as client:
    messages = client.get_messages(...)
```
and for async:
```python3
async with aleph_client.synchronous.Client() as client:
    messages = await client.get_messages(...)
```

**Comment by MHHukiewitz** [comment]
Then maybe we can work on integrating [AARS](https://github.com/aleph-im/active-record-sdk/) as part of this repository. It is quite lean, uses no extra dependencies compared to `aleph_client`, and can deliver this client wrapper.

I'm also currently working on automated pagination handling (as requested in #105) and [automated VM caching](https://github.com/aleph-im/active-record-sdk/blob/main/src/aars/core.py#L379). These features would be supported on the client object, but they still require pydantic objects to parse the response into. I could make this optional and allow regular dictionary retrieval with the client.

**Comment by odesenfans** [comment]
I don't see how AARS would help solve this? I think the last approach Hugo mentions makes the most sense, having an opaque client session object type that gets passed around and that contains the aiohttp client session.

**Comment by MHHukiewitz** [comment]
Like here?
https://github.com/aleph-im/active-record-sdk/blob/main/src/aars/core.py#L290

The only change necessary would be to make pydantic validation optional, then you could configure the session, API URL, whether to use caching, etc.

**Comment by odesenfans** [comment]
I'll take this issue, time to fix it.

**Comment by odesenfans** [comment]
Fixed by #140.

**Comment by MHHukiewitz** [comment]
Should be done now, as we are using AlephSession in aleph-sdk-python, or did I miss something?
